modyn/config/schema/pipeline/trigger/performance/performance.py

```diff
@@ -0,0 +1,95 @@
+from modyn.config.schema.pipeline.trigger.performance.criterion import (
+    PerformanceTriggerCriterion,
+)
+from modyn.const.types import ForecastingMethod, TriggerEvaluationMode
+
+
+class PerformanceTriggerEvaluationConfig(ModynBaseModel):
+    # max_samples_per_evaluation: int = Field(
+    #     1000,
+    #     description=(
+    #         "The maximum number of samples that should be used for the evaluation. If more are available, "
+    #         "downsampling will be used"
+    #     ),
+    # )  # TODO: currently not yet supported by the evaluator, we use a evaluation dataset that is sufficiently small
+
+    device: str = Field(description="The device the model should be put on.")
+    dataset: EvalDataConfig = Field(description="The dataset on which the model is evaluated.")
+    label_transformer_function: str | None = Field(
+        None, description="Function used to transform the label (tensors of integers)."
+    )
+
+    @field_validator("dataset")
+    @classmethod
+    def validate_metrics(cls, dataset: EvalDataConfig) -> EvalDataConfig:
+        """Assert that we have at least the accuracy metric."""
+        if not any(metric.name == "Accuracy" for metric in dataset.metrics):
+            raise ValueError("The accuracy metric is required for the performance trigger.")
+        return dataset
+
+
+class _InternalPerformanceTriggerConfig(ModynBaseModel):
+    detection_interval_data_points: int = Field(
+        description=(
+            "Specifies after how many samples another believe update (query density "
+            "estimation, accuracy evaluation) should be performed."
+        )
+    )
+    data_density_window_size: int = Field(
+        0,
+        description="The window size for the data density estimation. Only used for lookahead mode.",
+    )
+    performance_triggers_window_size: int = Field(
+        10,
+        description="The maximum number of evaluations after triggers to consider for computing the expect performance.",
+    )
+
+    evaluation: PerformanceTriggerEvaluationConfig = Field(
+        description="Configuration for the evaluation of the performance trigger."
+    )
+
+    decision_criteria: dict[str, PerformanceTriggerCriterion] = Field(
+        description=(
+            "The decision criteria to be used for the performance trigger. If any of the criteria is met, "
+            "the trigger will be executed. The criteria will be evaluated in the order they are defined. "
+            "Every criterion is linked to a metric. Some of the criteria implicitly only work on accuracy which is "
+            "the default metric that is always generated and cannot be disabled. To define a "
+            "`StaticPerformanceThresholdCriterion` on Accuracy, the evaluation config has to define the accuracy metric."
+            # TODO: maybe support custom aggregation functions
+        ),
+        min_length=1,
+    )
+
+    mode: TriggerEvaluationMode = Field(
+        "hindsight",
```

Comment:
As I probably have mentioned before, for the review of the actual algorithm, it will be helpful to have a README or something explaining this again. THanks!

Reply:
Okay, I'll open a separate documentation PR after the this here.

Reply:
#596